From 205f4abb9b980854e93ed58f4c2f245f63fe8116 Mon Sep 17 00:00:00 2001
From: Frederik Gladhorn <frederik.gladhorn@nokia.com>
Date: Mon, 27 Jun 2011 14:01:19 +0200
Subject: [PATCH] Notify about name/description changes for qml.

Declarative items would not emit the name/description/role
changed signals as they are supposed to.
Neither was updateAccessibility called.
Index: qt4-x11-4.7.4/src/declarative/accessible/qdeclarativeaccessible_p.h
===================================================================
--- qt4-x11-4.7.4.orig/src/declarative/accessible/qdeclarativeaccessible_p.h	2011-09-28 15:37:57.530973271 +0200
+++ qt4-x11-4.7.4/src/declarative/accessible/qdeclarativeaccessible_p.h	2011-09-28 15:38:08.707028685 +0200
@@ -236,13 +236,29 @@
     ~QDeclarativeAccessibleAttached();
 
     Role role() const { return (Role)m_role; }
-    void setRole(Role role) { m_role = (char)role; }
+    void setRole(Role role)
+    {
+        m_role = (char)role;
+        emit roleChanged();
+        // There is no way to signify role changes at the moment.
+        // QAccessible::updateAccessibility(parent(), 0, QAccessible::);
+    }
 
     QString name() const { return m_name; }
-    void setName(const QString &name) { m_name = name; }
+    void setName(const QString &name)
+    {
+        m_name = name;
+        emit nameChanged();
+        QAccessible::updateAccessibility(parent(), 0, QAccessible::NameChanged);
+    }
 
     QString description() const { return m_description; }
-    void setDescription(const QString &description) { m_description = description; }
+    void setDescription(const QString &description)
+    {
+        m_description = description;
+        emit descriptionChanged();
+        QAccessible::updateAccessibility(parent(), 0, QAccessible::DescriptionChanged);
+    }
 
     QDeclarativeItem *item() const {
         return static_cast<QDeclarativeItem*>(parent());
